library/src/main/java/it/sephiroth/android/library/tooltip/Tooltip.java

```diff
@@ -1010,6 +1058,37 @@ private void calculatePositions(List<Gravity> gravities, final boolean checkEdge
                 calculatePositionCenter(checkEdges, screenTop, width, height);
             }
 
+            // correction needed to support dialog as well e.g.
+            // => better solution would be to take this into account in calcualtions,
+            // but for now a post calculation correction must do it
+            if (mShowAsPopup) {
+                int offsetTop = 0;
+                int offsetLeft = 0;
+                int additionalOffsetLeft = 0;
+                int realStatusBarHeight = Utils.getStatusBarHeight(getContext());
+                int orientation = getResources().getConfiguration().orientation;
+                if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
+                    int softButtonsHeight = Utils.getSoftButtonsBarHeight(getContext());
+                    int deviceRotation = Utils.getDeviceRotation(getContext());
+                    // soft buttons are only at the left if device is rotated to the right
+                    if (softButtonsHeight > 0 && deviceRotation == Surface.ROTATION_270) {
+                        additionalOffsetLeft -= softButtonsHeight;
+                    }
+                }
+                mViewRect.top -= realStatusBarHeight;
+                mViewRect.bottom -= realStatusBarHeight;
+
+                Rect outRect = new Rect();
+                mView.getWindowVisibleDisplayFrame(outRect);
+                offsetTop -= outRect.top;
+                offsetLeft -= outRect.left;
+
+                mViewRect.left += offsetLeft + additionalOffsetLeft;
+                mViewRect.top += offsetTop;
+                mDrawRect.left += offsetLeft;
+                mDrawRect.top += offsetTop;
```

Comment:
That's why I wrote following:

> And there seems to be a very small offset in dialogs - I don't know to fix them yet, but it works at least and is accurate enough...

This is also the answer for your other questions, this is why I currently implemented the `asPopup()` method and did not change this to be the default implementation

Reply:
ok I missed that.
So I'd prefer to have this of a develop branch instead of the master branch, and continue working on that one until we figure out these things.
Do you agree?

Reply:
That's fine. I will use it in my app like this already as I need it and for now I'm fine with this. I could not find out yet why the position is slightly off nor have an idea currently though...

Reply:
Thank you! Would you mind create a new PR to merge into `develop`?

Reply:
Done, here: https://github.com/sephiroth74/android-target-tooltip/pull/110